Autopilot/boardfiles/Src/tim.c

```diff
@@ -95,6 +96,44 @@ void MX_TIM11_Init(void)
 
 }
 
+/* TIM12 Init Function */
+void MX_TIM12_Init(void)
+{
+  
+  TIM_IC_InitTypeDef sConfigIC = {0};
+
+  htim12.Instance = TIM12;
+  htim12.Init.Prescaler = 1;
+  htim12.Init.CounterMode = TIM_COUNTERMODE_UP;
+
+  //this period represents 1000hz - i.e every tick represents a millisecond
+  htim12.Init.Period = 0.001; 
```

Comment:
if you think back to the bootcamp, the period is the value that your timer will count until.

You want your counter to run at 1000 Hz, so that means your `htim12.Init.Period` should be 1000 if you want each tick to be a milisecond. 

Also, you need to set the prescalar correctly. Our CPU (STM32F765ZG) runs at 216 MHz, so your prescalar must be 216000

Reply:
Also not sure how this slipped my mind - I must've autopiloted.

Reply:
> Thanks for getting this in. I have a few comments.
> 
> Do you have a dev board that you can test this on?

Unfortunately I do not. Is there any way some else can test this for me?